lib/core/src/Cardano/Wallet/Primitive/Types/TokenPolicy.hs

```diff
@@ -278,8 +279,17 @@ validateMetadataURL = fmap AssetURL .
           | otherwise = Left $ "Scheme must be https: but got " ++ scheme
 
 validateMetadataUnit :: AssetUnit -> Either String AssetUnit
-validateMetadataUnit assetUnit@AssetUnit{name} =
-    (validateMinLength 1 name >>= validateMaxLength 30) $> assetUnit
+validateMetadataUnit = validateName >=> validateTicker >=> validateDecimals
+  where
+    validateName u@AssetUnit{name} =
+        (validateMinLength 1 name >>= validateMaxLength 30) $> u
+    validateTicker u@AssetUnit{ticker} =
+        case fmap validateMetadataTicker ticker of
+            Just (Left e) -> Left e
+            _ -> Right u
+    validateDecimals u@AssetUnit{decimals}
+        | decimals > 0 = Right u
```

Comment:
Note that the submitter also enforces that decimals are < 20

Reply:
Fixed, thanks.